### PR TITLE
Show Bangladesh National ID in patient entry screen

### PR DESCRIPTION
**Pull request by rakshakhegde**
https://www.pivotaltracker.com/story/show/168980840

### CODE REVIEW

File: app/src/test/java/org/simple/clinic/newentry/PatientEntryScreenLogicTest.kt [hunk 1]
```diff
@@ -166,6 +167,42 @@ class PatientEntryScreenLogicTest {
     verify(ui).openMedicalHistoryEntryScreen()
   }
 
+  @Test
+  fun `when bangladesh national id is entered then patient record should still be created`() {
```
Comment (sanchita) [inline app/src/test/java/org/simple/clinic/newentry/PatientEntryScreenLogicTest.kt]: Can we rename this test to be more specific?
`when bangladesh national id is entered and save is clicked then patient should get registered`

File: app/src/test/java/org/simple/clinic/newentry/PatientEntryScreenLogicTest.kt [hunk 2]
```diff
@@ -166,6 +167,42 @@ class PatientEntryScreenLogicTest {
...
+        personalDetails = PersonalDetails("Ashok", "12/04/1993", age = null, gender = Transgender),
+        address = Address(colonyOrVillage = "colony", district = "district", state = "state"),
+        phoneNumber = OngoingNewPatientEntry.PhoneNumber("1234567890"),
+        bangladeshNationalId = bangladeshNationalId
+    ))
+    verifyNoMoreInteractions(patientRepository)
+    verify(patientRegisteredCount).get()
+    verify(patientRegisteredCount).set(1)
+    verifyNoMoreInteractions(patientRegisteredCount)
+    verify(ui).openMedicalHistoryEntryScreen()
```
Comment (sanchita) [inline app/src/test/java/org/simple/clinic/newentry/PatientEntryScreenLogicTest.kt]: Love how water-tight this test is 👍

File: app/src/main/java/org/simple/clinic/patient/OngoingNewPatientEntry.kt [hunk 3]
```diff
@@ -96,6 +96,9 @@ data class OngoingNewPatientEntry(
   fun withConsent(reminderConsent: ReminderConsent): OngoingNewPatientEntry =
       copy(reminderConsent = reminderConsent)
 
+  fun withBangladeshNationalIdChanged(bangladeshNationalId: Identifier): OngoingNewPatientEntry =
```
[the remark this answers is not in the document]
Comment (rakshakhegde) [reply app/src/main/java/org/simple/clinic/patient/OngoingNewPatientEntry.kt]: Done

File: app/src/main/java/org/simple/clinic/newentry/PatientEntryModule.kt [hunk 4]
```diff
@@ -3,8 +3,8 @@ package org.simple.clinic.newentry
 import dagger.Module
```
[the remark this answers is not in the document]
Comment (rakshakhegde) [reply app/src/main/java/org/simple/clinic/newentry/PatientEntryModule.kt]: Shortened

File: app/src/main/java/org/simple/clinic/patient/OngoingNewPatientEntry.kt [hunk 5]
```diff
@@ -41,6 +41,7 @@ data class OngoingNewPatientEntry(
     val address: Address? = null,
     val phoneNumber: PhoneNumber? = null,
     val identifier: Identifier? = null,
+    val bangladeshNationalId: Identifier? = null,
```
[the remark this answers is not in the document]
Comment (rakshakhegde) [reply app/src/main/java/org/simple/clinic/patient/OngoingNewPatientEntry.kt]: I've tried making this change and found it is not as straightforward to update the model correctly when we convert this to a list of Business IDs. This needs a better approach to solve this issue which I think should be done in a separate PR.
Tracked this issue here: https://www.pivotaltracker.com/story/show/170401847

File: app/src/main/java/org/simple/clinic/patient/PatientRepository.kt [hunk 6]
```diff
@@ -328,6 +328,15 @@ class PatientRepository @Inject constructor(
           }
         }
 
+    val bangladeshNationalIdSave = cachedOngoingEntry
+        .flatMapCompletable { entry ->
+          if (entry.bangladeshNationalId == null) {
+            Completable.complete()
+          } else {
+            addIdentifierToPatient(patientUuid, entry.bangladeshNationalId, loggedInUser, facility).toCompletable()
+          }
```
[the remark this answers is not in the document]
Comment (rakshakhegde) [reply app/src/main/java/org/simple/clinic/patient/PatientRepository.kt]: Please refer the same discussion earlier: https://github.com/simpledotorg/simple-android/pull/990#discussion_r360406331

File: app/src/test/java/org/simple/clinic/newentry/PatientEntryScreenLogicTest.kt (hunk 1 above)
Comment (rakshakhegde) [reply app/src/test/java/org/simple/clinic/newentry/PatientEntryScreenLogicTest.kt]: Done

**Pull request merged**